syncer/syncer.go

```diff
@@ -1069,7 +1069,7 @@ func (s *Syncer) flushCheckPoints() error {
 	// optimistic shard info, DM-master may resolved the optimistic lock and let other worker execute DDL. So after this
 	// worker resume, it can not execute the DML/DDL in old binlog because of downstream table structure mismatching.
 	// We should find a way to (compensating) implement a transaction containing interaction with both etcd and SQL.
-	if err != nil {
+	if err != nil && (terror.ErrDBExecuteFailed.Equal(err) || terror.ErrDBUnExpect.Equal(err)) {
```

Comment:
Maybe only flushSafeModeExitPoint in we meet error?

Reply:
For etcd error I think it's still okay to flush global checkpoint. Because it's not saved.

Reply:
I'm afraid there will be some other error that leads to data loss.